content/en/blog/_posts/2021-07-16-read-write-once-pod-access-mode-alpha.md

```diff
@@ -0,0 +1,299 @@
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: ""
+```
+
+In Kubernetes, we currently have three access modes for PVs and PVCs:
+
+- ReadWriteOnce -- the volume can be mounted as read-write by a single node
+- ReadOnlyMany -- the volume can be mounted read-only by many nodes
+- ReadWriteMany -- the volume can be mounted as read-write by many nodes
+
+These access modes are enforced by Kubernetes components like the `kube-controller-manager` and `kubelet` to ensure only certain pods are allowed to access a given PersistentVolume.
+This allows users to follow the principle of least privilege to mitigate the impact of bad actors.
+
+## What is this new access mode and how does it work?
+
+In Kubernetes v1.22, a fourth access mode for PVs and PVCs is introduced for CSI volumes:
+
+- ReadWriteOncePod -- the volume can be mounted as read-write by a single pod
+
+When a user creates a pod with a PVC that has the ReadWriteOncePod access mode, Kubernetes will ensure that pod is the only pod in the cluster that can read and write to it.
+
+If you create another pod that references the same PVC with this access mode, the pod will fail to start like so:
+
+```
+Events:
+  Type     Reason            Age   From               Message
+  ----     ------            ----  ----               -------
+  Warning  FailedScheduling  1s    default-scheduler  0/1 nodes are available: 1 node has pod using PersistentVolumeClaim with the same name and ReadWriteOncePod access mode.
+```
+
+### How is this different than the ReadWriteOnce access mode?
+
+The ReadWriteOnce access mode restricts volume access to a single *node*, which means it is possible for multiple pods on the same node to read from and write to the same volume.
+This could potentially be a major problem for some applications, especially if they require at most one writer for data safety guarantees.
+
+With ReadWriteOncePod these issues go away.
+Just set the access mode on your PVC, and Kubernetes will guarantee only a single pod has access.
+
+## How do I use it?
+
+The ReadWriteOncePod access mode is in alpha for Kubernetes 1.22 and is only supported for CSI volumes.
+As a first step you need to enable the ReadWriteOncePod [feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates) for `kube-apiserver`, `kube-scheduler`, and `kubelet`.
+You can enable (or disable) the feature by setting command line arguments:
+
+```
+--feature-gates="...,ReadWriteOncePod=true"
+```
+
+### Creating a PersistentVolumeClaim
+
+In order to use the ReadWriteOncePod access mode for your PVs and PVCs, you will need to create a new PVC with the access mode:
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: top-secret
+spec:
+  accessModes:
+  - ReadWriteOncePod # Allow only a single pod to access top-secret.
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: ""
```

Comment:
I think you'll want to have 2 different PVC examples: one for dynamic provisioning, and one for pre-existing volumes. 

For dynamic provisioning, the storageClassName should be empty (to use the default storageclass) or filled in.

Reply:
Removed the `storageClassName` field.

This dynamic provisioning example is different (at least the name is), were there other changes I should make to differentiate?

Reply:
Renamed this to `single-writer-only` (from `top-secret`) to clarify the intended use case for the access mode